docs/proposals/26-workload-identity.md

```diff
@@ -0,0 +1,514 @@
+    "gardener.cloud": {
+        "workloadidentity": {
+            "name": "<workloadidentity-name>",
+            "namespace": "<workloadidentity-namespace>",
+            "uid": "<workloadidentity-uid>",
+        },
+        "shoot": {
+            "name": "<shoot-name>",
+            "namespace": "<shoot-namespace>",
+            "uid": "<shoot-uid>",
+        },
+        "project": {
+            "name": "<project-name>",
+            "uid": "<project-uid>",
+        },
+        "seed": {
+            "name": "<seed-name>",
+            "uid": "<seed-uid>",
+        },
+        "landscape": {
+            "id": "<gardener-landscape-id>",
+        },
+    }
+}
+```
+
+### Distribution of Workload Identity Tokens
+
+Gardenlet will request tokens as per the global configurations and renew them
+regularly. It will be responsible to provide information for the specific usage
+of the token, e.g. shoot name, namespace and UID, via the `TokenRequest` API.
+Only the `gardenlet` will be authorized to create and refresh workload identity
+tokens. Seed Authorizer will be extended to allow gardenlets to request workload
+identity tokens only for `WorkloadIdentity` that they are responsible for.
+
+As the tokens will usually have lifetime shorter than the period between two
+reconciliations, it is essential that the token creation and management are
+decoupled from the current control loops of gardenlet and implemented by a
+dedicated controller, also running in the gardenlet controller manager. Forced
+renewal of the tokens will be performed when the resource referring
+`WorkloadIdentity`s is annotated with
+`gardener.cloud/operation=renew-workload-identity-token`. The annotation is
+deliberately not set on the `WorkloadIdentity` because single `WorkloadIdentity`
+can be used by multiple shoots potentially running on different seeds, i.e.
+multiple controllers would be responsible to react on the annotation which is
+usually fine, but all of them would have to negotiate when the operation is
+completed and the annotation to be removed.
+
+To achieve the restriction only gardenlet to create or refresh tokens in the
+seed cluster, but not any other controller running there, a new CRD
+`WorkloadIdentityBinding` in the seed will be implemented. This CRD will refer
+to the `WorkloadIdentity` in the Garden cluster that is source of the token, as
+well as specify target store where the tokens to be written. Gardenlet will be
+the only component having write access to this CRD, while extension controllers
+and other components relying on workload identities should request no more than
+read access. This `WorkloadIdentityBinding` CRD will be reconciled by a
+dedicated controller named `workloadidentity-refresher` from Gardenlet
+controller manager responsible to request OIDC tokens and write them into the
+target store. The workload identity `spec.config` will be also written into the
+target store.
```

Comment:
TBH, I don't understand this entire paragraph. Can you please explain it again/rephrase? Independently of the content, it is a bit strange that a CRD in cluster A references another resource in cluster B. I don't think this is a typical pattern.

- In order to restrict gardenlet to be the only instance able to create/refresh tokens, isn't it enough to NOT provide the respective privilege to extensions? 
- What is a "target store"?